### PR TITLE
Fix proc timeout reached error detection

### PR DESCRIPTION
**Pull request by aliendrew**
Related to changes made in https://github.com/metarhia/metautil/commit/e4a521a49c0b16d02bfd42f06b2bed0a27d2bbed
- [x] tests and linter show no problems (`npm t`)
- [ ] tests are added/updated for bug fixes and new features
- [x] code is properly formatted (`npm run fmt`)
- [ ] description of changes is added in CHANGELOG.md
- [ ] update .d.ts typings

### CODE REVIEW

**Review by tshemsedinov — COMMENTED** [review]
Do you have bug reproduction test or code snippet?
Anyway solution is extremely inefficient

**Comment by aliendrew** [comment]
For example this api:
```js
({
  parameters: {
    a: 'number',
    b: 'number',
  },
  timeout: 100,
  method: async ({ a, b }) => {
    const result = a + b;
    await node.timers.promises.setTimeout(150);
    return result;
  },

  returns: 'number',
});
```
will log client error:
500     Error: Timeout of 100ms reached
but should be:
408     Error: Timeout of 100ms reached

**Review by aliendrew — COMMENTED** [review]
This solution should be better

File: lib/server.js [hunk 1]
```diff
@@ -266,8 +266,8 @@ class Server {
     try {
       result = await proc.invoke(context, args);
     } catch (error) {
-      if (error.message === 'Timeout reached') {
-        error.code = error.httpCode = 408;
+      if (error.code === 408) {
+        error.httpCode = error.code;
       }
       return void client.error(error.code, { id, error });
```
Comment (tshemsedinov) [inline lib/server.js:272]:
```suggestion
      let code = error.code === 'ETIMEOUT' ? 408 : 500;
      if (typeof error.code === 'number') code = error.code;
      error.httpCode = code;
      return void client.error(code, { id, error });
```

**Comment by tshemsedinov** [comment]
To be landed after https://github.com/metarhia/metautil/pull/195

FYI: @aliendrew see changes in this PR and PR mentioned above

**Pull request merged**